core/src/main/scala/kafka/log/Log.scala

```diff
@@ -320,7 +320,7 @@ class Log(@volatile private var _dir: File,
     initializeLeaderEpochCache()
     initializePartitionMetadata()
 
-    val nextOffset = loadSegments()
+    val nextOffset = loadSegments(hadCleanShutdown)
```

Comment:
There is no need to pass hadCleanShutdown in since it's already accessible from loadSegments().

Reply:
fixed.